x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/pagerduty/IncidentEvent.java

```diff
@@ -99,40 +100,85 @@ public HttpRequest createRequest(final String serviceKey, final Payload payload)
                 .scheme(Scheme.HTTPS)
                 .path(PATH)
                 .proxy(proxy)
+                .setHeader("Accept", ACCEPT_HEADER)
                 .jsonBody(new ToXContent() {
                     @Override
                     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-                        builder.field(Fields.SERVICE_KEY.getPreferredName(), serviceKey);
-                        builder.field(Fields.EVENT_TYPE.getPreferredName(), eventType);
-                        builder.field(Fields.DESCRIPTION.getPreferredName(), description);
+                        builder.field("routing_key", serviceKey);
+                        builder.field("event_action", eventType);
                         if (incidentKey != null) {
-                            builder.field(Fields.INCIDENT_KEY.getPreferredName(), incidentKey);
+                            builder.field("dedup_key", incidentKey);
                         }
+
+                        builder.startObject("payload");
+                        {
+                            builder.field("summary", description);
+
+                            if (attachPayload) {
```

Comment:
Just to make sure: This gets parsed properly on the other side if you leave it out (and dont have the `custom_details` structure)?

Reply:
custom_details is not mandatory, so it will def leave it out. Ill add a null check for payload as well in this statement